### DISCUSSION TITLE
Improve visibility into the cycle GC

### DISCUSSION THREAD

**Discussion by markshannon** [faster-cpython/ideas#259] [General]
The cycle GC takes up a significant part of the runtime of many Python programs. Quite a large part in some cases.

There are many ideas about how to improve it, but first of all we need some visibility into how it operates in practice.
To that end, we should add some stats, and some timings.

Things we would like to know, both for individual collections, and in total:
* Which generation is being collected (individual collections only).
* Number of objects scanned
* Number of objects collected
* Finalizers invoked
* Memory reclaimed
* Time spent

Anything else we should add to the above list?

**Comment by gvanrossum** [comment]
@nascheme ^^

**Comment by brandtbucher** [comment]
> Anything else we should add to the above list?

Number of objects untracked?

**Comment by kumaraditya303** [comment]
> Anything else we should add to the above list?

What kind of object forms the most ref cycles? Idea: When they are allocated, manually move them to older generation to reduce work done collecting in younger generations.
